jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -32,6 +37,78 @@ public void completeDiagnostic(Diagnostic diagnostic) {
 
     @Override
     public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
-
+    	if (unit != null) {
+    		try {
+    			IType[] alltypes = unit.getAllTypes(); 
+    			for (IType type : alltypes) {
+    				/* @PathParam Value Mismatch Warning */
+    				List<String> endpointPathVars = findAndProcessEndpointURI(type);
+    				if (endpointPathVars == null) {
+    					continue;
+    				}
+    				IMethod[] typeMethods = type.getMethods();
+    				for (IMethod method : typeMethods) {
+    					ILocalVariable[] methodParams = method.getParameters();
+    					for (ILocalVariable param : methodParams) {
+    						IAnnotation[] paramAnnotations = param.getAnnotations();
+    	    				for (IAnnotation annotation : paramAnnotations) {
+    	    					if (annotation.getElementName() == "PathParam") {
+    	    						IMemberValuePair[] valuePairs = annotation.getMemberValuePairs();
+    	    						for (IMemberValuePair pair : valuePairs) {
+    	    							if (pair.getMemberName().equals(WebSocketConstants.ANNOTATION_VALUE)) {
+    	    								String pathValue = (String) pair.getValue();
+    	    								if (!endpointPathVars.contains(pathValue)) {
+    	    									Diagnostic d = createPathParamWarningDiagnostic(annotation, unit);
+    	    									diagnostics.add(d);
+    	    								}
+    	    							}
+    	    						}
+    	    					}
+    	    				}
+    					}
+    				}
+    			}
+    		} catch (JavaModelException e) {
+    			JakartaCorePlugin.logException(WebSocketConstants.DIAGNOSTIC_ERR_MSG, e);
+    		}
+    	}
+    }
+    
+    private Diagnostic createPathParamWarningDiagnostic(IJavaElement annotation, IOpenable unit) throws JavaModelException {
+		ISourceRange nameRange = JDTUtils.getNameRange(annotation);
+		Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
+		Diagnostic diagnostic = new Diagnostic(range, WebSocketConstants.PATHPARAM_VALUE_WARN_MSG);
+		diagnostic.setSource(WebSocketConstants.DIAGNOSTIC_SOURCE);
+        diagnostic.setSeverity(WebSocketConstants.WARNING);
```

Comment:
Formatting is a bit off. Make sure you're using spaces

Reply:
Resolved